alfresco-transformer-base/src/test/java/org/alfresco/transformer/AbstractTransformerControllerTest.java

```diff
@@ -161,14 +162,26 @@ void generateTargetFileFromResourceFile(String actualTargetExtension, File testF
 
     protected File getTestFile(String testFilename, boolean required) throws IOException
     {
-        ClassLoader classLoader = getClass().getClassLoader();
-        URL testFileUrl = classLoader.getResource(testFilename);
-        if (required && testFileUrl == null)
+        if (System.getProperty("os.name").substring(0, 3).equals("Win"))
         {
-            throw new IOException("The test file " + testFilename +
-                                  " does not exist in the resources directory");
+            File testFileUrl = null;
+            try {
+                testFileUrl = new File(new File(getClass().getProtectionDomain().getCodeSource().getLocation().toURI()) + "\\" + testFilename);
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
+            }
+            return testFileUrl;
+        }
```

Comment:
Is there need to add a separate implementation for windows? Would this new implementation not work on linux too?

Reply:
Unfortunately it doesn't. Initially I also thought it would (see the first commit) but all tests on travis (runs linux VMs) failed with incorrect pathToFile.

Reply:
It looks like the cause of your error on travis was building the the end of the file path manually, specifically the use of a backslash. Would converting the URL in the original code to a URI not have the same affect without having to add multiple implementations?
`183             return testFileUrl == null ? null : new File(testFileUrl.toURI())`
I'm sure some additional logic may be required to catch the `URISyntaxException`.

Reply:
Also, in your current implementation, missing a file will produce a different error on windows as you only handle the `URISyntaxException`. A file not existing is specifically handled in the original code.

Reply:
I explored that option yesterday, but you can't surround a return statement in try/catch and if you add the exception to the method signature, you have to add the exception to all calling methods in the stack as well, which didn't seem like a good option. What you suggest wouldn't work without changing the original code a lot and I chose to split the logic. 
Your second point, the windows exception is different than linux, but I tested that scenario with no file and it seems like it's throwing the correct exception when a file is missing. I guess is either because both exception inherit from Exception or because IOException is already part of the method's signature.

Reply:
Ran through the suggested fixes, but they broke linux paths.